src/xdsl/dialects/builtin.py

```diff
@@ -214,8 +216,8 @@ def get_shape(self) -> List[int]:
     @builder
     def from_type_and_list(
             referenced_type: Attribute,
-            shape: List[Union[int, IntegerAttr]] = None) -> VectorType:
-        if shape is None:
+            shape: List[Union[int, IntegerAttr]] = []) -> VectorType:
```

Comment:
antipattern of mutable default argument. For type-checking use:
```
shape: Optional[List[Union[...]]
```
`typing.Optional[a]` is just sugar for `Union[NoneType, a]`